### PR TITLE
add flat option to _search

### LINKED ISSUE #388: extend the 'flat' option to Search and Raw REST services

### PR DESCRIPTION
**Pull request by MohamedHamouGisaia**
Fix #388

### CODE REVIEW

File: arlas-core/src/main/java/io/arlas/server/model/response/Hit.java [hunk 1]
```diff
@@ -38,8 +39,8 @@
     public Hit() {
     }
 
-    public Hit(CollectionReference collectionReference, Map<String, Object> source) throws ArlasException {
-        data = source;
+    public Hit(CollectionReference collectionReference, Map<String, Object> source, Boolean flat) throws ArlasException {
+        data = flat?MapExplorer.flat(source,new MapExplorer.ReduceArrayOnKey("_"), new HashSet<>()): source;
```
Comment (laurent-thiebaud-gisaia) [inline arlas-core/src/main/java/io/arlas/server/model/response/Hit.java]: Missing spaces around "?"

File: arlas-rest/src/main/java/io/arlas/server/rest/explore/raw/RawRESTService.java [hunk 2]
```diff
@@ -99,7 +106,7 @@ public Response getArlasHit(
             throw new NotFoundException("Document " + identifier + " not found.");
         }
 
-        Hit hit = new Hit(collectionReference, source);
+        Hit hit = new Hit(collectionReference, source, (flat!=null && flat));
```
Comment (laurent-thiebaud-gisaia) [inline arlas-rest/src/main/java/io/arlas/server/rest/explore/raw/RawRESTService.java]: If you prefer, there are more readable way to express it (pass a boolean and not a Boolean, or  Boolean.TRUE.equals(flat) )

File: arlas-core/src/main/java/io/arlas/server/model/response/Hit.java (hunk 1 above)
Comment (MohamedHamouGisaia) [reply arlas-core/src/main/java/io/arlas/server/model/response/Hit.java]: Done :+1:

File: arlas-rest/src/main/java/io/arlas/server/rest/explore/raw/RawRESTService.java (hunk 2 above)
Comment (MohamedHamouGisaia) [reply arlas-rest/src/main/java/io/arlas/server/rest/explore/raw/RawRESTService.java]: Done :+1:

**Pull request merged**